app/controllers/recommendations_controller.rb

```diff
@@ -22,8 +30,57 @@ def create
     end
   end
 
+  def review_form
+    @recommendation = @recommendations.last
+
+    respond_to do |format|
+      format.html
+    end
+  end
+
+  def review
+    respond_to do |format|
+      @recommendation.assign_attributes(recommendation_params)
+      @recommendation.review!
+
+      format.html do
+        redirect_to @planning_application, notice: "Recommendation was successfully reviewed."
+      end
+    end
+  end
+
   private
 
+  def set_planning_application_with_recommendations
+    planning_application = planning_applications_scope.find(planning_application_id)
+
+    @planning_application = PlanningApplicationPresenter.new(view_context, planning_application)
+  end
+
+  def set_recommendations
+    @recommendations = @planning_application.recommendations
+  end
+
+  def planning_applications_scope
+    current_local_authority.planning_applications.includes(:recommendations)
+  end
+
+  def planning_application_id
+    Integer(params[:planning_application_id])
```

Comment:
Isn't Rails clever enough to not need this? `PlanningApplication.find("1")` should return the same as `PlanningApplication.find(1)`

Reply:
habit and i think Andrew told me once to do this before.. let me ask why again 😄

Reply:
thanks...good to know!